src/js/hyperschedule.js

```diff
@@ -1669,21 +1669,66 @@ function downloadPDF()
           course.courseName,
           width - 12,
         );
+        const courseLocationLines = pdf.splitTextToSize(
+          slot.location,
+          width - 12,
+        );
+        const courseInstructorsLines = pdf.splitTextToSize(
+          course.faculty.map(fullName => fullName.split(",")[0]).join(", "),
+          width - 12
+        );
 
+        // Attributes to be shown are calculated based on space available
+        // order of preference: Code, location, name, professor
+        // order of display: code, name, professor, location
+        const maxTextLength = (yEnd - yStart) - 2 * pdf.getLineHeight();
+        const toLocationLength = courseCodeLines.length +
+              courseLocationLines.length;
+        const toNameLength = toLocationLength + courseNameLines.length;
+        const toInstructorLength = toNameLength + courseInstructorsLines.length;
+        const showLocation = toLocationLength * pdf.getLineHeight() <
```

Comment:
Edge case to allow 50 minute blocks to contain most locations. Not sure how comment heavy you like your code, but could have a comment here or just get rid of this.

Reply:
Why does this differ from the other cases below?

Reply:
Also, in general, more comments is almost never a bad thing.